### PR TITLE
Move argo fake client to Tekton fake client. Also update apiserver, common, and crd tests

### PR DESCRIPTION
**Pull request by Tomcli**
**Which issue is resolved by this Pull Request:** 
Related #388

**Description of your changes:**
Move argo fake client to Tekton fake client and update apiserver, common, and crd tests. Many of the removed tests are heavily typed with Argo. Pending the followings to completely remove argo client.
- agent (Evan working on it)
- cache (Peng is making changes, can temporary comment out all the argo section for now.)
- crd <del>(some test codes are not matching, will do a follow up PR to match with the unit tests)</del> (done)

Also added new Travis job to run the new unit tests without argo.

**Environment tested:**

* Python Version (use `python --version`):
* Tekton Version (use `tkn version`):
* Kubernetes Version (use `kubectl version`):
* OS (e.g. from `/etc/os-release`):

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

   PR titles examples:
    * `fix(frontend): fixes empty page. Fixes #1234`
       Use `fix` to indicate that this PR fixes a bug.
    * `feat(backend): configurable service account. Fixes #1234, fixes #1235`
       Use `feat` to indicate that this PR adds a new feature. 
    * `chore: set up changelog generation tools`
       Use `chore` to indicate that this PR makes some changes that users don't need to know.
    * `test: fix CI failure. Part of #1234`
        Use `part of` to indicate that a PR is working on an issue, but shouldn't close the issue when merged.

- [ ] Do you want this pull request (PR) cherry-picked into the current release branch?
    
    If yes, use one of the following options:
  
    * **(Recommended.)** Ask the PR approver to add the `cherrypick-approved` label to this PR. The release manager adds this PR to the release branch in a batch update.
    *  After this PR is merged, create a cherry-pick PR to add these changes to the release branch. (For more information about creating a cherry-pick PR, see the [Kubeflow Pipelines release guide](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#option--git-cherry-pick).)

### CODE REVIEW

**Comment by Tomcli** [comment]
@ckadner the valid link test keeps failing because one of the twitter links is rate limited. Is there a way we could bypass that?

**Comment by ckadner** [comment]
> @ckadner the valid link test keeps failing because one of the twitter links is rate limited. Is there a way we could bypass that?

@Tomcli -- I created PR #396 to relax the Markdown link verification to 404 errors only. Ignoring rate-limited URLs.

File: .travis.yml [hunk 1]
```diff
@@ -65,6 +65,11 @@ matrix:
       go: 1.15.x
       script:
         - make build-backend
+    - name: "Verify go unit tests"
```
Comment (ckadner) [inline .travis.yml:68]: I think this should be called "Run go unit tests"? The tests are not _subject_ to verification, the tests are _doing_ the verification, no?

File: backend/src/common/util/scheduled_workflow_test.go [hunk 2]
```diff
@@ -18,13 +18,14 @@ import (
...
 	"time"
 
-	workflowapi "github.com/argoproj/argo/pkg/apis/workflow/v1alpha1"
 	swfapi "github.com/kubeflow/pipelines/backend/src/crd/pkg/apis/scheduledworkflow/v1beta1"
 	"github.com/stretchr/testify/assert"
 	core "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// Removed Argo spec test
```
Comment (ckadner) [inline backend/src/common/util/scheduled_workflow_test.go:27]: should this comment move down to line 117?

**Review by ckadner — CHANGES_REQUESTED** [review]
Hi @Tomcli -- thanks for adding comments where large code blocks were removed to help with future merges of the upstream KFP code. There are a few large blocks that still miss a comment, could you add those? Use a common comment (same wording) that we can search for so we may be able to exclude those reported merge conflicts in the future

File: backend/src/apiserver/resource/resource_manager_test.go [hunk 3]
```diff
@@ -126,28 +121,6 @@ func initWithJob(t *testing.T) (*FakeClientManager, *ResourceManager, *model.Job
 	return store, manager, j
 }
 
-func initWithOneTimeRun(t *testing.T) (*FakeClientManager, *ResourceManager, *model.RunDetail) {
```
Comment (ckadner) [inline backend/src/apiserver/resource/resource_manager_test.go:129]: add a comment why it was removed to help with future KFP merges?

File: backend/src/apiserver/resource/resource_manager_test.go [hunk 4]
```diff
@@ -170,77 +143,6 @@ func initWithPatchedRun(t *testing.T) (*FakeClientManager, *ResourceManager, *mo
 	return store, manager, runDetail
 }
 
-func initWithOneTimeFailedRun(t *testing.T) (*FakeClientManager, *ResourceManager, *model.RunDetail) {
```
Comment (ckadner) [inline backend/src/apiserver/resource/resource_manager_test.go:173]: add a comment why it was removed to help with future KFP merges?

File: backend/src/apiserver/resource/resource_manager_test.go [hunk 5]
```diff
@@ -664,91 +233,6 @@ func TestCreateRun_InvalidWorkflowSpec(t *testing.T) {
 	assert.Contains(t, err.Error(), "Failed to unmarshal workflow spec manifest")
 }
 
-func TestCreateRun_NullWorkflowSpec(t *testing.T) {
```
Comment (ckadner) [inline backend/src/apiserver/resource/resource_manager_test.go:667]: add a comment why it was removed to help with future KFP merges? the one above might not show in the Git conflict resolution

File: backend/src/apiserver/resource/resource_manager_test.go [hunk 6]
```diff
@@ -758,28 +242,6 @@ func TestDeleteRun_RunNotExist(t *testing.T) {
 	assert.Contains(t, err.Error(), "not found")
 }
 
-func TestDeleteRun_CrdFailure(t *testing.T) {
```
Comment (ckadner) [inline backend/src/apiserver/resource/resource_manager_test.go:761]: add a comment why it was removed to help with future KFP merges?

File: backend/src/apiserver/resource/resource_manager_test.go [hunk 7]
```diff
@@ -824,39 +286,14 @@ func TestDeleteExperiment_ExperimentNotExist(t *testing.T) {
...
 	store, manager, experiment := initWithExperiment(t)
 	defer store.Close()
 
-	manager.argoClient = client.NewFakeArgoClientWithBadWorkflow()
+	store.DB().Close()
 	err := manager.DeleteExperiment(experiment.UUID)
-	assert.Nil(t, err)
-}
-
-func TestDeleteExperiment_DbFailure(t *testing.T) {
```
Comment (ckadner) [inline backend/src/apiserver/resource/resource_manager_test.go:836]: add a comment why it was removed to help with future KFP merges?

File: backend/src/apiserver/resource/resource_manager_test.go [hunk 8]
```diff
@@ -868,32 +305,6 @@ func TestTerminateRun_RunNotExist(t *testing.T) {
 	assert.Contains(t, err.Error(), "not found")
 }
 
-func TestTerminateRun_DbFailure(t *testing.T) {
```
Comment (ckadner) [inline backend/src/apiserver/resource/resource_manager_test.go:871]: add a comment why it was removed to help with future KFP merges?

File: backend/src/apiserver/resource/resource_manager_test.go [hunk 9]
```diff
@@ -903,26 +314,6 @@ func TestRetryRun_RunNotExist(t *testing.T) {
 	assert.Contains(t, err.Error(), "not found")
 }
 
-func TestRetryRun_FailedDeletePods(t *testing.T) {
```
Comment (ckadner) [inline backend/src/apiserver/resource/resource_manager_test.go:906]: add a comment why it was removed to help with future KFP merges?

File: backend/src/apiserver/resource/resource_manager_test.go [hunk 10]
```diff
@@ -953,287 +344,6 @@ func TestCreateJob_ThroughWorkflowSpec(t *testing.T) {
 	assert.Equal(t, expectedJob, job)
 }
 
-func TestCreateJob_ThroughPipelineID(t *testing.T) {
```
Comment (ckadner) [inline backend/src/apiserver/resource/resource_manager_test.go:956]: add a comment why it was removed to help with future KFP merges?

File: backend/src/apiserver/resource/resource_manager_test.go [hunk 11]
```diff
@@ -1301,1126 +411,6 @@ func TestDeleteJob_DbFailure(t *testing.T) {
...
-	// Delete the above pipeline_version.
-	err = manager.DeletePipelineVersion(FakeUUIDOne)
-	assert.NotNil(t, err)
-
-	// Verify the version in deleting status.
-	version, err := manager.pipelineStore.GetPipelineVersionWithStatus(FakeUUIDOne, model.PipelineVersionDeleting)
-	assert.Nil(t, err)
-	assert.NotNil(t, version)
-}
-
```
Comment (ckadner) [inline backend/src/apiserver/resource/resource_manager_test.go:2423]: add a comment why it was removed to help with future KFP merges?

File: backend/src/apiserver/resource/resource_manager_util_test.go [hunk 12]
```diff
@@ -29,6 +28,8 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+// remove argo spec test
```
Comment (ckadner) [inline backend/src/apiserver/resource/resource_manager_util_test.go:31]: move this comment to line 150?

File: backend/src/apiserver/server/pipeline_server_test.go [hunk 13]
```diff
@@ -83,181 +25,10 @@ func TestCreatePipeline_InvalidYAML(t *testing.T) {
 	resourceManager := resource.NewResourceManager(clientManager)
 
 	pipelineServer := PipelineServer{resourceManager: resourceManager, httpClient: httpServer.Client()}
-	_, err := pipelineServer.CreatePipeline(context.Background(), &api.CreatePipelineRequest{
```
Comment (ckadner) [inline backend/src/apiserver/server/pipeline_server_test.go:86]: add a comment why it was removed to help with future KFP merges?

File: backend/src/apiserver/server/pipeline_upload_server_test.go [hunk 14]
```diff
@@ -145,317 +144,3 @@ func TestUploadPipeline_YAML(t *testing.T) {
...
-	encodedName := url.PathEscape(
-		"this is a loooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooog name")
-	req, _ = http.NewRequest("POST", "/apis/v1beta1/pipelines/upload_version?name="+encodedName+"&pipelineid="+resource.DefaultFakeUUID, bytes.NewReader(b.Bytes()))
-	req.Header.Set("Content-Type", w.FormDataContentType())
-	rr = httptest.NewRecorder()
-	handler = http.HandlerFunc(server.UploadPipelineVersion)
-	handler.ServeHTTP(rr, req)
-	assert.Equal(t, 400, rr.Code)
-	assert.Contains(t, string(rr.Body.Bytes()), "Pipeline name too long")
-}
```
Comment (ckadner) [inline backend/src/apiserver/server/pipeline_upload_server_test.go:461]: add a comment why it was removed to help with future KFP merges?

File: backend/src/apiserver/server/util_test.go [hunk 15]
```diff
@@ -108,90 +84,6 @@ func TestDecompressPipelineZip_MalformedZip2(t *testing.T) {
 	assert.Contains(t, err.Error(), "Not a valid zip file")
 }
 
-func TestDecompressPipelineZip_NonYamlZip(t *testing.T) {
```
Comment (ckadner) [inline backend/src/apiserver/server/util_test.go:111]: add a comment why it was removed to help with future KFP merges? I know there is a comment above, but that might not show in Git merge conflict resolution scope.

File: backend/src/common/util/formatter_test.go [hunk 16]
```diff
@@ -316,32 +289,12 @@ func TestFormatEmptyWorkflow(t *testing.T) {
...
 
-	expected := &v1alpha1.Workflow{
+	expected := &v1beta1.PipelineRun{
 		ObjectMeta: v1.ObjectMeta{GenerateName: "workflow-"}}
 
 	err := formatter.Format(workflow)
 	assert.Nil(t, err)
 	assert.Equal(t, expected, workflow)
 }
-
```
Comment (ckadner) [inline backend/src/common/util/formatter_test.go:328]: add a comment why it was removed to help with future KFP merges?

File: backend/src/common/util/workflow_test.go [hunk 17]
```diff
@@ -135,42 +137,23 @@ func TestWorkflow_ScheduledAtInSecOr0(t *testing.T) {
...
 	// No status
-	workflow = NewWorkflow(&workflowapi.Workflow{
-		Status: workflowapi.WorkflowStatus{},
+	workflow := NewWorkflow(&workflowapi.PipelineRun{
+		Status: workflowapi.PipelineRunStatus{},
 	})
 	assert.Equal(t, "", workflow.Condition())
 }
 
-func TestToStringForStore(t *testing.T) {
```
Comment (ckadner) [inline backend/src/common/util/workflow_test.go:153]: add a comment why it was removed to help with future KFP merges?

File: backend/src/common/util/workflow_test.go [hunk 18]
```diff
@@ -179,53 +162,8 @@ func TestWorkflow_OverrideName(t *testing.T) {
...
-					{Name: "PARAM3", Value: StringPointer("NEW_VALUE3")},
-					{Name: "PARAM4", Value: StringPointer("NEW_VALUE4")},
-					{Name: "PARAM5", Value: StringPointer("")},
-				},
-			},
-		},
-	}
-	assert.Equal(t, expected, workflow.Get())
-}
-
```
Comment (ckadner) [inline backend/src/common/util/workflow_test.go:226]: add a comment why it was removed to help with future KFP merges?

File: backend/src/common/util/workflow_test.go [hunk 19]
```diff
@@ -253,50 +191,16 @@ func TestWorkflow_SetOwnerReferences(t *testing.T) {
 	assert.Equal(t, expected, workflow.Get())
 }
 
-func TestWorkflow_SetLabelsToAllTemplates(t *testing.T) {
```
Comment (ckadner) [inline backend/src/common/util/workflow_test.go:256]: add a comment why it was removed to help with future KFP merges?

File: backend/src/common/util/workflow_test.go [hunk 20]
```diff
@@ -305,234 +209,3 @@ func TestSetLabels(t *testing.T) {
 
 	assert.Equal(t, expected, workflow.Get())
 }
-
```
Comment (ckadner) [inline backend/src/common/util/workflow_test.go:308]: add a comment why it was removed to help with future KFP merges?

File: backend/src/crd/controller/scheduledworkflow/client/workflow_client_test.go [hunk 21]
```diff
@@ -18,60 +18,17 @@ import (
...
-		CreatedAt:   metav1.NewTime(time.Unix(50, 0).UTC()),
-		StartedAt:   metav1.NewTime(time.Unix(51, 0).UTC()),
-		FinishedAt:  metav1.NewTime(time.Unix(52, 0).UTC()),
-		ScheduledAt: metav1.NewTime(time.Unix(54, 0).UTC()),
-		Index:       55,
-	}
-
-	assert.Equal(t, []swfapi.WorkflowStatus{*expected}, result)
-}
-
```
Comment (ckadner) [inline backend/src/crd/controller/scheduledworkflow/client/workflow_client_test.go:70]: add a comment why it was removed to help with future KFP merges?

File: backend/src/crd/controller/scheduledworkflow/client/workflow_client_test.go [hunk 22]
```diff
@@ -164,56 +121,3 @@ func TestRetrieveIndex(t *testing.T) {
...
-	assert.Nil(t, err)
-	expected = expected.Add(*req)
-
-	req, err = labels.NewRequirement(commonutil.LabelKeyWorkflowIndex, selection.GreaterThan,
-		[]string{"50"})
-	assert.Nil(t, err)
-	expected = expected.Add(*req)
-
-	assert.Equal(t, expected, *result)
-}
```
Comment (ckadner) [inline backend/src/crd/controller/scheduledworkflow/client/workflow_client_test.go:219]: add a comment why it was removed to help with future KFP merges?

File: backend/src/crd/controller/scheduledworkflow/util/scheduled_workflow_test.go [hunk 23]
```diff
@@ -529,152 +528,3 @@ func TestScheduledWorkflow_GetNextScheduledEpoch_UpdateStatus_WithWorkflow(t *te
...
-				Parameters: []workflowapi.Parameter{
-					{Name: "PARAM1", Value: commonutil.StringPointer("NEW_VALUE1_19700101100000")},
-					{Name: "PARAM2", Value: commonutil.StringPointer("NEW_VALUE2_1")},
-				},
-			},
-		},
-	}
-
-	assert.Equal(t, expected, result.Get())
-}
```
Comment (ckadner) [inline backend/src/crd/controller/scheduledworkflow/util/scheduled_workflow_test.go:680]: add a comment why it was removed to help with future KFP merges?

**Comment by ckadner** [comment]
If we use the same wording for all those comments, then we can search for those the next time we merge upstream KFP code and we may be able to exclude those reported merge conflicts automatically

**Comment by animeshsingh** [comment]
/hold

until we come back in Jan before merging

File: .travis.yml (hunk 1 above)
Comment (Tomcli) [reply .travis.yml]: done

**Comment by evan-hataishi** [comment]
/lgtm

I'll can add in the agent unit test command to the makefile when this is merged.

**Review by ckadner — APPROVED** [review]
Thanks for adding the comments Tommy

/lgtm

**Comment by animeshsingh** [comment]
/unhold

@Tomcli  @ckadner do we have a good handle being able to relicate these changes as we pull in latest ?

**Comment by ckadner** [comment]
> @Tomcli @ckadner do we have a good handle being able to replicate these changes as we pull in latest ?

I did a try run of merging in the latest changes from KFP 1.3.0 into master it shows me 895 files changed with **43 rejected conflicts**.

If I attempt the same KFP merge on top of this PR I get **56 rejected conflicts**. So roughly 30% more work for KFP merges going forward.

**Pull request merged**